### PR TITLE
feat(http): set by default Accept header for json requests

### PR DESCRIPTION
**Pull request by secustor**
## Changes
Adds by default an `Accept` header to http calls originating from `getJson`. 

## Context
Came across this while debugging https://github.com/renovatebot/renovate/pull/19173 and the backend seems to require this. 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

**Review by viceice — COMMENTED** [review]
Isn't that already set by got? i thought `responseType=json` should do that. 🤔

**Comment by secustor** [comment]
Seems like it actually should... 🤔     
https://github.com/sindresorhus/got/blob/fd7536078148461ad9cc72f2d57f89c98ca8f63e/source/core/index.ts#L634

Got on this trail because I have got some 406 errors on `https://releases.hashicorp.com`.

**Comment by viceice** [comment]
> responseType

no, you're on wrong version. We're still using v11.
https://github.com/sindresorhus/got/blob/5e17bb748c260b02e4cf716c2f4079a1c6a7481e/source/as-promise/index.ts#L196

this was the only line were the accept is set 😕

File: lib/util/http/index.ts [hunk 1]
```diff
@@ -245,6 +245,11 @@ export class Http<Opts extends HttpOptions = HttpOptions> {
       method,
       responseType: 'json',
     };
+    // signal that we expect a json response
+    opts.headers = {
+      Accept: 'application/json',
```
Comment (viceice) [inline lib/util/http/index.ts:250]:
```suggestion
      accept: 'application/json',
```

should be lowercased

**Review by viceice — COMMENTED** [review]
otherwise lgtm

Should maybe have a test-case 😏

File: lib/util/http/index.ts (hunk 1 above)
Comment (viceice) [reply lib/util/http/index.ts:250]: image

**Review by viceice — APPROVED** [review]

**Pull request merged**